nilearn/plotting/tests/test_find_cuts.py

```diff
@@ -52,7 +52,7 @@ def test_find_cut_coords():
     data = np.zeros((20, 20, 20))
     data[4:6, 4:6, 4:6] = 1000
     img = nibabel.Nifti1Image(data, 2 * np.eye(4))
-    mask_data = np.ones((20, 20, 20), dtype=int)
+    mask_data = np.ones((20, 20, 20), dtype="uint8")
```

Comment:
There seems to be a lot of tests throughout the library that build their own example images by creating an array and affine and then creating the `Nifti1Image`. the plotting test utils also provide fixtures for this. It would be better to rely on the utilities in `_utils.data_gen` to avoid all this duplication. I can think of 2 reasons why we keep duplicating this code rather than using the existing functions:
- we forget or don't know they exist -> we can add a note about this in the user guide; maybe rename the file? (`_utils` sounds like utils for the library itself but maybe fake data generation should go in something called `testing_utils`, we also have `_utils/testing.py`); PR reviewers should remember to check if code that already is available is not being duplicated
- the functions in `data_gen` may be a bit too complex; for simple cases re-writing the image generation is faster than reading their docstrings. Maybe we need to add a smaller simple function to cover frequent and simple use cases.

Reply:
fixtures could definitely help for tests that only need one default image; for more complex scenarios I bet importing `data_gen` and calling a function would feel simpler than parametrizing a pytest fixture. I completely agree that copy-pasting the data-generation snippets from other tests is very natural and that re-writing those would help; in the meanwhile paying attention to this in code reviews and encouraging the use of `data_gen` would be a start